src/cljs/ui/component/button.cljs

```diff
@@ -2,16 +2,19 @@
   (:require [shared.components :refer [button]]))
 
 (defn- file-select-handler [{:keys [file-with-meta file-accept-predicate on-file-accepted on-file-rejected]}]
-  (if (file-accept-predicate file-with-meta)
-    (let [{:keys [file]}      file-with-meta
-          array-buffer-reader (js/FileReader.)]
-      (set! (.-onload array-buffer-reader) (fn [^js e]
-                                             (let [data (-> e .-target .-result)]
-                                               (on-file-accepted (merge file-with-meta
-                                                                        {:data data})))))
-      (.readAsArrayBuffer array-buffer-reader file))
-    (when on-file-rejected
-      (on-file-rejected))))
+  (-> (file-accept-predicate file-with-meta)
```

Comment:
how do we know that `file-accept-predicate` is a Promise? Both cases should be handled, or at the very least there should be an assertion that checks if it's an instance of `js/Promise`

Reply:
well it is not a library, it is the type that file-accept-predicate should return from now on. We can just add that to the component doc string. I think it doesn't worth the trouble of making a fn that can handle both cases, is not that other people is going to use that.
But agree that it needs to be documented.

Reply:
just added a docstring to the component, take a look and see if it works for you. If you feel strong about supporting both cases we can open a issue so we don't block more important work.

Reply:
I was thinking `(assert (instanceOf js/promise file-accept-predicate))` but docstring is enough as well